backend/src/db/services/user.service.ts

```diff
@@ -102,4 +106,52 @@ export class UserService {
         throw new ConflictException('Email is already taken.')
     }
   }
+
+  async updateUserPaymentPlan(
+    stripeCustomerId: string,
+    paymentPlan: PaymentOptions,
+  ) {
+    try {
+      await this.userModel.findOneAndUpdate(
+        { stripeCustomerId },
+        { paymentPlan },
+      )
+    } catch (error) {
+      this.logger.error(error)
+      throw new UnprocessableEntityException(
+        'Could not update payment plan of the provided customer from Stripe',
+      )
+    }
+  }
+
+  async updateUserCheckoutInformation(
+    stripeCustomerId: string,
+    checkoutInformation: CheckoutInformation,
+  ) {
+    // We do 2 calls here since we want to know if the customerId exists
+    const user = await this.userModel.findOne({ stripeCustomerId })
+    if (!user)
+      throw new UnprocessableEntityException(
+        'Could not match Stripe event to any user',
+      )
+    try {
+      await this.userModel.findOneAndUpdate(
+        {
+          stripeCustomerId,
+          'checkoutInformation.lastInformationTime': {
+            $lt: checkoutInformation.lastInformationTime,
+          },
+        },
+        { checkoutInformation },
+        { new: true },
+      )
+    } catch (error) {
+      this.logger.error(error)
+      throw new ServiceUnavailableException(error)
```

Comment:
Sanitize errors, dont pass them to the client directly. In this case
```suggestion
      throw new ServiceUnavailableException("Something went wrong, please try again later!")
```